packages/flutter_tools/test/commands.shard/hermetic/test_test.dart

```diff
@@ -59,6 +59,29 @@ void main() {
     Cache: () => FakeCache(),
   });
 
+  testUsingContext('Pipes shard-index and total-shards to package:test',
+      () async {
+    final FakePackageTest fakePackageTest = FakePackageTest();
+
+    final TestCommand testCommand = TestCommand(testWrapper: fakePackageTest);
+    final CommandRunner<void> commandRunner =
+        createTestCommandRunner(testCommand);
+
+    await commandRunner.run(const <String>[
+      'test',
+      '--total-shards=1',
+      '--shard-index=2',
+      '--no-pub',
+    ]);
+
+    expect(fakePackageTest.lastArgs, contains('--total-shards=1'));
+    expect(fakePackageTest.lastArgs, contains('--shard-index=2'));
```

Comment:
This test makes sense. Can you also add a test that verifies that `--total-shards` and `--shard-index` are NOT piped through to `fakePackageTest` if they're not provided to the `commandRunner`?